src/mixins/DrawingTool.js

```diff
@@ -135,10 +135,13 @@ const DrawingTool = types
           return value;
         }, { coordstype: 'px', dynamic: self.dynamic });
 
-        const newArea = self.annotation.createResult(value, currentArea.results[0].value.toJSON(), control, obj);
+        const [main, ...rest] = currentArea.results;
+        const newArea = self.annotation.createResult(value, main.value.toJSON(), control, obj);
+
+        rest.forEach(r => newArea.addResult(r.toJSON()));
```

Comment:
it's better to add comment why do we do this, all this behaviour is not very obvious and common